InformationScripting/src/queries/CompositeQuery.cpp

```diff
@@ -110,6 +110,17 @@ QList<Optional<TupleSet> > CompositeQuery::execute(QList<TupleSet> input)
 	return outNode_->calculatedOutputs_;
 }
 
+Query* CompositeQuery::addQuery(std::unique_ptr<Query> query)
+{
+	auto rawQ = query.get();
+	// A query should only be added once:
+	Q_ASSERT(std::find_if(nodes_.begin(), nodes_.end(), [rawQ](QueryNode* existing) {return existing->q_.get() == rawQ;})
+					== nodes_.end());
+	auto newNode = new QueryNode(std::move(query));
```

Comment:
Hm I used this answer http://stackoverflow.com/a/8114913 as a guideline. It advises to use value semantics whenever we take ownership. On the other hand to this answer there is a comment which suggests to use &&. Should I still use &&?